### PR TITLE
[15.0][MIG]stock_account_prepare_anglo_saxon_out_lines_hook

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
The first hook is not needed anymore, the other it is.
@JoanMForgeFlow

### CODE REVIEW

**Review by JoanMForgeFlow — APPROVED** [review]
Great! :+1:

File: stock_account_prepare_anglo_saxon_out_lines_hook/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from odoo import models
+
+
+class AccountMove(models.Model):
+
+    _inherit = "account.move"
+
+    def _prepare_interim_account_line_vals(self, line, move, debit_interim_account):
+        # Compute accounting fields.
+        sign = -1 if move.type == "out_refund" else 1
```
Comment (MiquelRForgeFlow) [inline stock_account_prepare_anglo_saxon_out_lines_hook/models/account_move.py]:
```suggestion
        sign = -1 if move.move_type == "out_refund" else 1
```

File: stock_account_prepare_anglo_saxon_out_lines_hook/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            "debit": balance < 0.0 and -balance or 0.0,
+            "credit": balance > 0.0 and balance or 0.0,
+            "account_id": debit_interim_account.id,
+            "exclude_from_invoice_tab": True,
+            "is_anglo_saxon_line": True,
+        }
+
+    def _prepare_expense_account_line_vals(self, line, move, credit_expense_account):
+        # Compute accounting fields.
+        sign = -1 if move.type == "out_refund" else 1
```
Comment (MiquelRForgeFlow) [inline stock_account_prepare_anglo_saxon_out_lines_hook/models/account_move.py]:
```suggestion
        sign = -1 if move.move_type == "out_refund" else 1
```

**Review by MiquelRForgeFlow — COMMENTED** [review]
please, backport this fixes to v14

**Review by MiquelRForgeFlow — APPROVED** [review]
I have to say that I don't see the neediness of this module, as the monkey patch now is redundant.

If you want to customize the values of `_prepare_interim_account_line_vals` and `_prepare_expense_account_line_vals`, you can already do it by calling `super` to `_stock_account_prepare_anglo_saxon_out_lines_vals`. To know if its one type or the other, you can check if they have `analytic_account_id` in the dictionary or not.

**Comment by AaronHForgeFlow** [comment]
Yes, I can inherit from _stock_account_prepare_anglo_saxon_out_lines_vals but then I would need to do the same loop and and put all the ifs statements because the checks are the same. That would make the inheritance unefficient. And what is the drawback of adding this? Just for beauty?

**Comment by AaronHForgeFlow** [comment]
Closing as not popular :cry:

**Comment by AaronHForgeFlow** [comment]
Reopened until Odoo accepts a hook for better inheritance.

**Comment by rvalyi** [comment]
@AaronHForgeFlow out of curiosity: do you guys use the anglo-saxon mode in Spain?

**Comment by AaronHForgeFlow** [comment]
just created the PR to Odoo here: https://github.com/odoo/odoo/pull/98947 if that is merge this module is not needed.

Also fixed the commit history in this PR, that was incorrect.

@rvalyi  HI! :) honestly, I don't know but i don't think so. I did this for a US company and I know it is used in Denmark as well.

**Comment by AaronHForgeFlow** [comment]
Closing, as there are a solution without the hook. The inheritance is not as easy as a prepare_method but it is possible. The module is deprecated. Sorry for the noise.

**Comment by rvalyi** [comment]
Hello @AaronHForgeFlow that would be appreciated if you put some link to some module or even a snippet where you override the method.

**Comment by MiquelRForgeFlow** [comment]
@rvalyi you can compare https://github.com/OCA/account-financial-tools/tree/14.0/account_move_line_sale_info with https://github.com/OCA/account-financial-tools/tree/15.0/account_move_line_sale_info as an example.

**Comment by rvalyi** [comment]
Thanks. We are using these kind enumerate/index hacks too in some other places BTW :-)

**Pull request closed without merging**